checkbox-ng/checkbox_ng/launcher/master.py

```diff
@@ -166,6 +166,41 @@ def invoked(self, ctx):
         else:
             print(_("\nConnection timed out."))
 
+    def check_remote_api_match(self):
+        """
+        Check that agent and controller are running on the same
+        REMOTE_API_VERSION else exit checkbox with an error
+        """
+        agent_api_version = self.sa.get_remote_api_version()
+        controller_api_version = RemoteSessionAssistant.REMOTE_API_VERSION
+        if agent_api_version == controller_api_version:
+            return
+        template_msg = (
```

Comment:
I think it is reasonable, it is quite a long message with just small variations, it would be very ugly to have two separate messages.

Reply:
The problem is that the person reading this will have to redo the work of an interpreter. As for argument for having two strings means you have to keep them in sync -> this template has a certain number of placeholders that have to be kept in sync with the format call.